Loop/Managers/LoopAppManager.swift

```diff
@@ -0,0 +1,307 @@
+//
+//  LoopAppManager.swift
+//  Loop
+//
+//  Created by Darin Krauss on 2/16/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+import LoopKit
+import LoopKitUI
+
+public protocol ViewControllerProvider: AnyObject {
+    var viewController: UIViewController? { get set }
+}
+
+class LoopAppManager: NSObject {
+    private enum State: Int {
+        case initialize
+        case launchManagers
+        case launchOnboarding
+        case launchHomeScreen
+        case launchComplete
+
+        var next: State { State(rawValue: rawValue + 1) ?? .launchComplete }
+    }
+
+    private var launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+    private var window: UIWindow?
+
+    private var pluginManager: PluginManager!
+    private var bluetoothStateManager: BluetoothStateManager!
+    private var alertManager: AlertManager!
+    private var loopAlertsManager: LoopAlertsManager!
+    private var trustedTimeChecker: TrustedTimeChecker!
+    private var deviceDataManager: DeviceDataManager!
+    private var onboardingManager: OnboardingManager!
+
+    private var state: State = .initialize
+
+    private let log = DiagnosticLog(category: "LoopAppManager")
+
+    // MARK: - Initialization
+
+    func initialize(with launchOptions: [UIApplication.LaunchOptionsKey: Any]?) {
+        dispatchPrecondition(condition: .onQueue(.main))
+        precondition(state == .initialize)
+
+        self.launchOptions = launchOptions
+
+        registerBackgroundTasks()
+
+        self.state = state.next
+    }
+
+    func launch(into window: UIWindow?) {
+        dispatchPrecondition(condition: .onQueue(.main))
+        precondition(!isLaunchComplete)
+        precondition(state != .initialize)
+
+        guard isProtectedDataAvailable() else {
+            log.default("Protected data not available; deferring launch...")
+            return
+        }
+
+        self.window = window
+        
+        window?.tintColor = .loopAccent
```

Comment:
[nit] I'm curious what this is for, but not blocking for sure...

Reply:
It was in `AppDelegate` so I moved it here. It was added by @nhamming for https://github.com/tidepool-org/Loop/pull/184 related to https://tidepool.atlassian.net/browse/LOOP-309.